_episodes/06-loops.md

```diff
@@ -210,6 +210,26 @@ Note that the loop variable is a macro, not a scalar. This helps us write code w
 > {: .solution}
 {: .challenge}
 
+
+> ## Challenge
+> Write a do-file named "append-gdp-all.do" that appends all data which name matches the pattern gdp`year'.dta. Generate a variable called year that records the gdp year as indicated in the name of the file. Label the variables accordingly. Save the final dataset as gdp1990-2017.dta".
+> > ## Solution
+> > ```
+> > use "data/gdp1990.dta", clear
+> > generate year=1990
+> > label variable gdp_per_capita "GDP per capita"
+> > label variable year "Year"
+> >  forvalues year = 1991/2017 {
+> >      append using "data/gdp`year'.dta"
+> >      replace year=`year' if missing(year)
```

Comment:
Also here.

Reply:
Btw what's the point of 
. forvalues i = 1/5 {
  2.     generate gdp_`i' = gdp_per_capita^`i'
  3. }
(9,381 missing values generated)
(9,381 missing values generated)
(9,381 missing values generated)
(9,381 missing values generated)
(9,381 missing values generated)

. su gdp_?

 is it to reinforce the use of wildcards? Can the summarize command go inside the loop?